contracts/arbos/builtin/ArbSys.sol

```diff
@@ -1,31 +1,59 @@
 pragma solidity >=0.4.21 <0.7.0;
 
+/**
+* @title Precompiled contract that exists in every Arbitrum chain at address(100), 0x0000000000000000000000000000000000000064. Exposes a variety of system-level functionality.
+ */
 interface ArbSys {
-    // Get ArbOS version number
+    /**
+    * @notice  Get ArbOS version number
+    * @return version number int
+     */
     function arbOSVersion() external pure returns (uint);
 
-    // Get Arbitrum block number
+    /**
+    * @notice Get Arbitrum block number (Arbitrum genisis starts at 0, increments with qqq)
```

Comment:
When it increments is a bit complicated.  It increments whenever the Ethereum block number or timestamp of a transaction is greater than the previous transaction.  But the sequencer can also decide to cause it to be incremented even if the L1 blocknum / timestamp doesn't change.   What we can guarantee is that all txs within the same Arbitrum blocknum will have the same Ethereum blocknum and timestamp.

Reply:
Yeah this will be documented elsewhere, maybe here just best to clarify that it's distinct from the L1 block number